lib/mainloop-control.c

```diff
@@ -409,6 +410,48 @@ export_config_graph(ControlConnection *cc, GString *command, gpointer user_data)
   control_connection_send_reply(cc, result);
 }
 
+static void
+control_internal_logs(ControlConnection *cc, GString *command, gpointer user_data)
+{
+  GString *result = g_string_new("");
+  gchar **arguments = g_strsplit(command->str, " ", 0);
+  AFInterLive ret;
+
+  if (g_str_equal(arguments[1], "START"))
+    {
+      ret = afinter_start_live_collection();
+
+      switch (ret)
+        {
+        case AFINTER_LIVE_COLLECTION_STARTED:
+          g_string_assign(result, "OK Started to collect internal messages");
+          break;
+
+        case AFINTER_LIVE_COLLECTION_RUNNING:
+          g_string_assign(result, "FAIL Error: Live collection of internal logs is already running");
+          break;
+
+        case AFINTER_INTERNAL_SRC_PRESENT:
+          g_string_assign(result, "FAIL Error: Internal source is already configured");
+          break;
+
+        default:
+          break;
+        }
+    }
+  else if (g_str_equal(arguments[1], "STOP"))
+    {
+      afinter_stop_live_collection();
+      afinter_get_collected_messages(result);
```

Comment:
I get assertion error with the following case:
```
@version: 3.25

log {
    source { example-msg-generator(num(1)); };
    destination { file(/dev/stdout); };
};
```

Start syslog-ng with `./syslog-ng -Fe -f ../etc/internal.conf`.

After calling start, and then stop, syslog-ng emits: 
```
[2020-03-03T13:05:58.048621] g_queue_is_empty: assertion 'queue != NULL' failed
```

Reply:
@furiel : I tried this scenario and did not receive the error in the new version.